### PR TITLE
Replace Python emulator with Go emulator

### PR DESCRIPTION
**Pull request by hannesmann**
Closes: #63

### CODE REVIEW

File: go.sum
Comment (alekodu) [inline go.sum:1]: Is this file needed as part of the repo or should we add it to the gitignore file?

File: model/istio.go
Comment (alekodu) [inline model/istio.go:1]: Are there any dependencies on this file? I think this was part of an older design that may not be longer needed...

File: model/redis.go
Comment (alekodu) [inline model/redis.go:1]: Similarly with this file, I think we don't use anything with Redis anymore; but if we use some of the functionalities there, we should probably change the name of this file.

File: generator/deploy.sh
[the remark this answers is not in the document]
Comment (hannesmann) [reply generator/deploy.sh:1]: I only changed the permissions on this file. It wasn't marked as executable so I couldn't run it.

**Comment by hannesmann** [comment]
> I think you removed many files with functionality that may be needed later on but I guess that you can bring back as you work on those features... is that the idea?

Yes, I'm going to add the functionality incrementally in several pull requests. I'm using the old Python emulator in the main branch as a reference. 

> Also as I mentioned before, it would be better if we find a way to keep modularity of the two core functionalities, the generator and the emulator, before merging the changes.

I'll see if I can get it working with a Go workspace instead. If I understand correctly, we should keep the old application-generator module and add two new ones for the model and emulator?

**Comment by alekodu** [comment]
> > I think you removed many files with functionality that may be needed later on but I guess that you can bring back as you work on those features... is that the idea?
> 
> Yes, I'm going to add the functionality incrementally in several pull requests. I'm using the old Python emulator in the main branch as a reference.
> 
> > Also as I mentioned before, it would be better if we find a way to keep modularity of the two core functionalities, the generator and the emulator, before merging the changes.
> 
> I'll see if I can get it working with a Go workspace instead. If I understand correctly, we should keep the old application-generator module and add two new ones for the model and emulator?

I think so. It would be good to have all common code in a module that is then used by both the generator and the model (emulator) part.

**Comment by hannesmann** [comment]
I pushed a commit that separates the code into three modules:
application-model (for the common code)
application-generator
application-emulator

I wasn't sure if it should be called application-model or something else (like application-common) since the emulator is the model of the application. If we want to change the names I'll make a new commit, otherwise I think this is ready to be merged.
